src/auth/src/credentials/idtoken/impersonated.rs

```diff
@@ -144,12 +161,56 @@ impl Builder {
     }
 
     /// Should include email claims in the ID Token.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use google_cloud_auth::credentials::idtoken;
+    /// # use serde_json::json;
+    /// # let impersonated_credential = json!({
+    /// #     "type": "impersonated_service_account",
+    /// #     "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/test-principal:generateAccessToken",
+    /// #     "source_credentials": {
+    /// #         "type": "authorized_user",
+    /// #         "client_id": "test-client-id",
+    /// #         "client_secret": "test-client-secret",
+    /// #         "refresh_token": "test-refresh-token"
+    /// #     }
+    /// # });
+    /// let audience = "https://my-service.a.run.app";
+    /// let credentials = idtoken::impersonated::Builder::new(audience, impersonated_credential)
+    ///     .with_include_email()
+    ///     .build();
+    /// // Now you can use credentials.id_token().await to fetch the token.
+    /// ```    
     pub fn with_include_email(mut self) -> Self {
         self.include_email = Some(true);
         self
     }
 
     /// Sets the chain of delegates.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use google_cloud_auth::credentials::idtoken;
+    /// # use serde_json::json;
+    /// # let impersonated_credential = json!({
+    /// #     "type": "impersonated_service_account",
+    /// #     "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/test-principal:generateAccessToken",
+    /// #     "source_credentials": {
+    /// #         "type": "authorized_user",
+    /// #         "client_id": "test-client-id",
+    /// #         "client_secret": "test-client-secret",
+    /// #         "refresh_token": "test-refresh-token"
+    /// #     }
+    /// # });
+    /// let audience = "https://my-service.a.run.app";
+    /// let credentials = idtoken::impersonated::Builder::new(audience, impersonated_credential)
```

Comment:
Here and the previous examples, consider at least making the type visible:


```suggestion
    /// let audience = "https://my-service.a.run.app";
    /// let credentials = idtoken::impersonated::Builder::new(audience, impersonated_credentials("test-principal"))

    /// fn impersonated_credentials(principal: &str) -> serde_json::Value {
    /// serde_json::json!({
    ///         "type": "impersonated_service_account",
    ///         "service_account_impersonation_url": format!( "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/{principal}:generateAccessToken"),
    ///       "source_credentials": {
    ///         // See https://docs.cloud.google.com/iam/docs/keys-create-delete#iam-service-account-keys-create-console for possible formats
    ///        }
    /// })
    /// }
```